openid-4-verifiable-presentations-1_0.md

```diff
@@ -616,11 +622,322 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+this object are defined per Credential Format in (#format_specific_properties). If omitted,
+no specific constraints are placed on the metadata or validity of the requested
+Credential.
+
+`claims`:
+: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
+claims in the requested Credentials.
+
+`claim_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for
+elements in `claims`. The rules for selecting claims to send are defined in (#selecting_claims).
+
+Note that multiple Credential Queries in a request MAY request a presentation of the same Credential.
+
+## Credential Set Query {#credential_set_query}
+
+A Credential Set Query is an object representing a request for one or more credentials to satisfy
+a particular use case with the Verifier.
+
+Each entry in `credential_sets` MUST be an object with the following properties:
+
+`options`
+: REQUIRED: A non-empty array, where each value in the array represents a distinct query that satisfies 
+the use case. The value of each element in the `options` array is an array of identifiers which reference
+elements in `credentials`.
+
+`required`
+: OPTIONAL. A boolean flag which indicates whether this set of Credentials is required
+to satisfy the particular use case at the Verifier.
+If omitted, the effective value MUST be processed as `true`.
+
+`purpose`
+: OPTIONAL. A string, number or object specifying the purpose of the
+query. This specification does not define a specific structure or specific values
+for this property. The purpose is intended to be used by the Verifier to
+communicate the reason for the query to the Wallet. The Wallet MAY use this
+information to show the user the reason for the request.
+
+## Claims Query {#claims_query}
+
+Each entry in `claims` MUST be an object with the following properties:
+
+`id`:
+: REQUIRED if `claim_sets` is present in the Credential Query; OPTIONAL otherwise. A string
+identifying the particular claim. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the particular `claims` array, the same `id` MUST NOT
+be present more than once.
+
+`path`:
+: REQUIRED if the Credential Format uses a JSON-based claims structure; MUST NOT
+be present otherwise. The value MUST be a claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+`namespace`:
+: REQUIRED if the Credential Format is based on the mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the data element
+within the mdoc, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the data element identifier of the data element within the provided namespace
+in the mdoc, e.g., `first_name`.
+
```

Comment:
```suggestion
`conditional`:
: OPTIONAL. A boolean that can be used to indicate that an element is only required if it is present in the credential. 
If the value is present, the RP indicates if the element is present in the credential, it is required to be returned.
Note that because from a security perspective the RP cannot rely on the wallet to be truthful about the presence of data elements in the credential, this indicator can only be meaningfully used if the RP knows through an out-of-band mechanism whether a credential must have certain element or not. 
```

Add conditional elements as an explicit mechanism instead of relying on mandating the wallet to adhere to the order of preference.